src/middlewares/checkAdmin.ts

```diff
@@ -0,0 +1,36 @@
+import {
+  Injectable,
+  CanActivate,
+  ExecutionContext,
+  Inject,
+  NotFoundException
+} from '@nestjs/common';
+import { UserRepository } from '../../src/users/repository/userRepository';
+import * as admin from 'firebase-admin';
+
+@Injectable()
+export class RolesGuard implements CanActivate {
+  @Inject()
+  usersRepository: UserRepository;
+
+  async canActivate(context: ExecutionContext): Promise<boolean> {
+    try {
+      const req = context.switchToHttp().getRequest();
+      const token = req.headers.authorization.split(' ')[1];
```

Comment:
Can we use `res.locals.userUid` Instead code duplication? Please follow DRY principles